mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -116,6 +129,9 @@ abstract class MLWriter extends BaseReadWrite with Logging {
   }
 
   // override for Java compatibility
+  override def session(sparkSession: SparkSession): this.type = super.session(sparkSession)
+
+  // override for Java compatibility
   override def context(sqlContext: SQLContext): this.type = super.context(sqlContext)
```

Comment:
Just realized: Can you please change this to call ```super.session(sqlContext.sparkSession)``` to avoid a deprecation warning?  Same for the other call to context() below.